services/galley/src/Galley/API/Teams.hs

```diff
@@ -420,32 +422,45 @@ updateTeamMember zusr zcon tid targetMember = do
       >>= permissionCheck SetMemberPermissions
   -- user may not elevate permissions
   targetPermissions `ensureNotElevated` user
-  -- target user must be in same team
   Data.teamMember tid targetId >>= \case
-    Nothing -> throwM teamMemberNotFound
-    _ -> pure ()
-  -- cannot demote only owner (effectively removing the last owner)
-  members <- Data.teamMembersUnsafeForLargeTeams tid
-  okToDelete <- canBeDeleted members targetId tid
-  when (not okToDelete && targetPermissions /= fullPermissions) $
-    throwM noOtherOwner
+    Nothing -> do
+      -- target user must be in same team
+      throwM teamMemberNotFound
+    Just previousMember -> do
+      when (downgradesOwner previousMember targetPermissions)
+        $
+        -- owner can be downgraded IFF it can be deleted
+        unless (canDeleteMember user previousMember)
+        $ throwM accessDenied
   -- update target in Cassandra
   Data.updateTeamMember tid targetId targetPermissions
-  let otherMembers = filter (\u -> u ^. userId /= targetId) members
-      updatedMembers = targetMember : otherMembers
-  -- note the change in the journal
-  when (team ^. teamBinding == Binding) $ Journal.teamUpdate tid updatedMembers
-  -- inform members of the team about the change
-  -- some (privileged) users will be informed about which change was applied
-  let privileged = filter (`canSeePermsOf` targetMember) updatedMembers
-      mkUpdate = EdMemberUpdate targetId
-      privilegedUpdate = mkUpdate $ Just targetPermissions
-      privilegedRecipients = membersToRecipients Nothing privileged
-  now <- liftIO getCurrentTime
-  let ePriv = newEvent MemberUpdate tid now & eventData ?~ privilegedUpdate
-  -- push to all members (user is privileged)
-  let pushPriv = newPush zusr (TeamEvent ePriv) $ privilegedRecipients
-  for_ pushPriv $ \p -> push1 $ p & pushConn .~ Just zcon
+  (updatedMembers, tooMany) <- Data.teamMembers' tid Nothing
```

Comment:
If we don't use `teamMembersUnsafeForLargeTeams` we risk sending wrong data to billing and the fanout. We have decided to not fanout things to larger teams, but can we please implement that when we're implementing that?